app/models/user.rb

```diff
@@ -545,10 +506,13 @@ def banished?
   end
 
   def subscribe_to_mailchimp_newsletter
+    # in app/services/mailchimp/bot.rb, the user object has no setting or
+    # notification_setting records without calling this hook within this method
+    create_users_settings_and_notification_settings_records
```

Comment:
Should these be created in a `before_create` hook instead so that once the user exists they definitely have both settings models? Profile creation could potentially also move in there instead of being done in an `after_create_commit` hook.

Reply:
Yes, I discussed this approach at length with Dan Uber, but did not implement in my [closed PR](https://github.com/forem/forem/pull/13904) cos I wasn't sure how to also do the same to Profile. I'll teach myself from Molly's implementation here 😅